pages/api/play-attempt/index.ts

```diff
@@ -0,0 +1,100 @@
+import { LevelModel, PlayAttemptModel, StatModel } from '../../../models/mongoose';
+import withAuth, { NextApiRequestWithAuth } from '../../../lib/withAuth';
+import { NextApiResponse } from 'next';
+import { ObjectId } from 'bson';
+import PlayAttempt from '../../../models/schemas/playAttemptSchema';
+import dbConnect from '../../../lib/dbConnect';
+import getTs from '../../../helpers/getTs';
+
+const MINUTE = 60;
+
+export default withAuth(async (req: NextApiRequestWithAuth, res: NextApiResponse) => {
+  if (req.method !== 'POST') {
+    return res.status(405).json({
+      error: 'Method not allowed',
+    });
+  }
+
+  if (!req.body) {
+    return res.status(400).json({
+      error: 'Missing required parameters',
+    });
+  }
+
+  const { levelId } = req.body;
+
+  if (!levelId) {
+    return res.status(400).json({
+      error: 'Missing required parameters',
+    });
+  }
+
+  await dbConnect();
+  const now = getTs();
+  // first don't do anything if user has already beaten this level
+  const [level, playAttempt, statRecord] = await Promise.all([
+    LevelModel.findById(levelId),
+    PlayAttemptModel.findOneAndUpdate({
+      userId: req.user._id,
+      levelId: levelId,
+      endTime: { $gt: now - 15 * MINUTE },
+    }, {
+      $set: {
+        endTime: now,
+      },
+      $inc: { updateCount: 1 }
+    }, {
+      new: true,
+      sort: { endTime: -1 }
+    }),
+    StatModel.findOne({
+      userId: req.user._id,
+      levelId: levelId,
+    }),
+  ]);
+
+  if (!level || level.isDraft) {
+    return res.status(404).json({
+      error: 'Level not found',
+    });
+  }
+
+  if (playAttempt) {
+    if (statRecord?.complete) {
```

Comment:
Should we be updating the endTime in this case? I like the idea of having a separate end time (completedTime?), so that extra play after completing the level doesn't skew the data incorrectly.

Reply:
Can you elaborate? If someone beat the level that completed time would be on the stat record wouldnt it?

What do you mean that the data get skewed?

Reply:
Forgot about the timestamp within stats...